couch/tests/conftest.py

```diff
@@ -84,82 +115,83 @@ def send_replication(couch_version):
+        print("[INFO] create replication task {}".format(i))
+        body = replication_body.copy()
+        body['_id'] = 'my_replication_id_{}'.format(i)
+        body['target'] = body['target'] + str(i)
+        r = requests.post(
+            replicator_url,
+            auth=(common.NODE1['user'], common.NODE1['password']),
+            headers={'Content-Type': 'application/json'},
+            json=body,
+        )
+        r.raise_for_status()
+        print("[INFO] replication task created:", r.json())
+
+
+def get_replication():
     """
     Attempt to get active replication tasks
     """
-    if couch_version == '1':
-        return
-
     task_url = "{}/_active_tasks".format(common.NODE1['server'])
 
     r = requests.get(task_url, auth=(common.NODE1['user'], common.NODE1['password']))
     r.raise_for_status()
-    count = len(r.json())
+    active_tasks = r.json()
+    print("[INFO] active_tasks response:", active_tasks)
+    count = len(active_tasks)
     return count > 0
 
 
-def generate_data(couch_version):
-    """
-    Generate data on the couch cluster to test metrics.
-    """
-    # pass in authentication info for version 2
-    auth = (common.USER, common.PASSWORD) if couch_version == "2" else None
+def enable_cluster():
+    auth = (common.USER, common.PASSWORD)
     headers = {'Accept': 'text/json'}
 
-    # Generate a test database
-    requests.put("{}/kennel".format(common.URL), auth=auth, headers=headers)
-
-    # Populate the database
-    data = {
-        "language": "javascript",
-        "views": {
-            "all": {"map": "function(doc) { emit(doc._id); }"},
-            "by_data": {"map": "function(doc) { emit(doc.data, doc); }"},
-        },
-    }
-    requests.put("{}/kennel/_design/dummy".format(common.URL), json=data, auth=auth, headers=headers)
-
-    urls = [
-        "{}/_node/node1@127.0.0.1/_stats".format(common.URL),
-        "{}/_node/node2@127.0.0.1/_stats".format(common.URL),
-        "{}/_node/node3@127.0.0.1/_stats".format(common.URL),
-    ]
-
-    ready = defaultdict(bool)
-    for _ in range(120):
-        print("Waiting for stats to be generated on the nodes...")
-        try:
-            for url in urls:
-                if not ready[url]:
-                    res = requests.get(url, auth=auth, headers=headers)
-                    if res.json():
-                        ready[url] = True
-            if len(ready) and all(ready.values()):
-                break
-        except Exception:
-            pass
-        sleep(1)
-
-    if couch_version == "1":
-        return
-
-    doc_url = "{}/_replicator/_all_docs".format(common.URL)
-    for _ in range(120):
-        try:
-            res = requests.get(doc_url, auth=auth, headers=headers)
-            data = res.json()
-            if data.get('rows'):
-                break
-        except Exception:
-            pass
-        sleep(1)
+    requests_data = []
+    for node in ["couchdb-1.example.com", "couchdb-2.example.com"]:
```

Comment:
Should we use the updated `NODE_N` dicts?

```suggestion
    for node in [NODE1, NODE2]:
        remote_node = node["name"]
```

Reply:
Good catch. thx.